chr-data/src/CHR/Data/Lens/MicroLens.hs

```diff
@@ -91,19 +91,19 @@ a ^. f = get f a
 
 -- | Alias for 'get' to avoid conflict with state get; not happy choice because of 'getl'
 getL :: (f :-> a) -> f -> a
-getL = flip (^.)
+getL = \x y -> y ^. x
```

Comment:
why are these eta expansions necessary?

Reply:
They are due to the [simplified subsumption](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0287-simplify-subsumption.rst) changes in GHC 9.0.1.

Reply:
Although I see now that I only really need one eta-expansion for `getL`, this works too: `\x -> (^. x)`.

Reply:
In this case:

```
flip (^.) :: ((a -> Const a a) -> s -> Const a s) -> s -> a
-- but
getL :: (forall f. Functor f => (a -> f a) -> s -> f s) -> s -> a
```

So, to match these two the `f` would need to be instantiated to `Const a`, but the `forall f.` is not at the top-level, so this instantiation is deep and no longer allowed since 9.0.1.

At least that is my understanding.